src/type.cpp

```diff
@@ -1349,6 +1381,8 @@ bool ArrayType::IsIntType() const { return false; }
 
 bool ArrayType::IsUnsignedType() const { return false; }
 
+bool ArrayType::IsSignedType() const { return false; }
```

Comment:
I don't understand why `ArrayType::IsSignedType` is not derived from element type like it is done, e.g., for `VectorType`.

Reply:
Array is an array, it's not signed and not unsigned, not fp and not int. It's an array. I.e. if the request if it's array of signed types, then appropriate request needs to be made. But the one cannot make conclusions like "it's signed int type, so I can do integer arithmetic on it, load or store it" out of this check, which would be misleading.